pkg/infra/iac2/templates/kubernetes_provider/factory.ts

```diff
@@ -0,0 +1,15 @@
+import * as pulumi_k8s from '@pulumi/kubernetes'
+
+interface Args {
+    Name: string
+    KubeConfig: string
+}
+
+// noinspection JSUnusedLocalSymbols
+function create(args: Args): pulumi_k8s.Provider {
+    return new pulumi_k8s.Provider(args.Name, {
+        kubeconfig: args.KubeConfig,
+    })
```

Comment:
Right. I didn't even bother with the contents of KubeConfig since those rely on the cluster resource and AWS region and it's not clear what patterns we should be using here.